core/Translate.php

```diff
@@ -123,7 +133,15 @@ private static function getTranslator()
 
     public static function loadAllTranslations()
     {
+        self::triggerDeprecationNotice();
         self::loadCoreTranslation();
         Manager::getInstance()->loadPluginTranslations();
     }
+
+    protected static function triggerDeprecationNotice()
+    {
+        if (Development::isEnabled()) {
+            trigger_error('Using \\Piwik\\Translate is deprecated. Use \\Piwik\\Translation\\Translator instead.', E_USER_DEPRECATED);
```

Comment:
btw any reason we're not using Log::warn or so?

Reply:
No. Guess a warning should be fine as well, as it's only in development mode